### PR TITLE
Add allowed values for list options

### PR DESCRIPTION
**Pull request by jablko**
Extend the current behavior with respect to `option.type` to `option.element.type` as well. This e.g. gets the allowed values for list options, analogous to other options. The only current affected option is `lib`: This change keeps its allowed values up to date.

### CODE REVIEW

File: packages/tsconfig-reference/scripts/tsconfigRules.ts [hunk 1]
```diff
@@ -3,6 +3,24 @@ import * as remark from "remark";
...
+    | "object"
+    | "list"
+    | Map<string, number | string>;
+  defaultValueDescription?: string | number | boolean | ts.DiagnosticMessage;
+  element: CommandLineOption;
+}
+
+declare module "typescript" {
+  const optionDeclarations: CommandLineOption[];
+  const optionsForWatch: CommandLineOption[];
```
Comment (jablko) [inline packages/tsconfig-reference/scripts/tsconfigRules.ts:21]: Declaring these skeleton types means we can drop some runtime checks/assertions, below.

File: packages/tsconfig-reference/scripts/tsconfigRules.ts [hunk 2]
```diff
@@ -173,26 +191,17 @@ function trueIf(name: string) {
   ];
 }
 
-// Number/boolean options without explicit default are 0/false
-const defaultDefaults = { number: "0", boolean: "false" };
```
Comment (jablko) [inline packages/tsconfig-reference/scripts/tsconfigRules.ts:177]: Merged -> https://github.com/microsoft/TypeScript/pull/46280/commits/4981c560171d94e1d7f88d78d2bf41f8ea0e612d

File: packages/tsconfig-reference/scripts/tsconfigRules.ts [hunk 3]
```diff
@@ -238,31 +247,47 @@ export const defaultsForOptions = {
...
-            synonyms.length > 1
-              ? synonyms.map((name) => `\`${name}\``).join("/")
-              : synonyms[0]
-          ),
-        ];
-      })
+    [...ts.optionDeclarations, ...ts.optionsForWatch].map((option) => [
+      option.name,
+      formatAllowedValues(
+        option.type === "list" ? option.element.type : option.type
```
Comment (jablko) [inline packages/tsconfig-reference/scripts/tsconfigRules.ts:270]: Extend current `option.type` behavior -> `option.element.type`.

**Comment by orta** [comment]
Structurally yep I like this, I think I need to do some design work to reduce the line height on it to get it merged 👍🏻 

Screen Shot 2021-10-26 at 12 47 18 PM

**Comment by orta** [comment]
With ES2022 this list gets another 6 values, so maybe the sidebar doesn't scale here - will sit down and have a proper thing

**Comment by jablko** [comment]
:+1: Makes sense, and no rush. Is this mergeable with the latest revision: Restore the `lib.md` and hide the `lib` allowed values? What's left is the refactoring, without actually affecting any current options.

**Comment by orta** [comment]
It is 👍🏻

**Pull request merged**